src/org/exist/xmldb/RemoteCollection.java

```diff
@@ -647,10 +647,12 @@ private void uploadAndStore(final Resource res) throws XMLDBException {
                             fileName = (String) xmlRpcClientLease.get().execute("uploadCompressed", params);
                         }
                     }
-                } else {
+                }
+
+                if (fileName == null) {
```

Comment:
`fileName` is set in line 642 or line 647 ...  which checking for null? probably there is an indenting confusion for me...